lib/tools/system-calls.js

```diff
@@ -216,4 +217,178 @@ systemCallMethods.getPortFromEmulatorString = function (emStr) {
   return false;
 };
 
+systemCallMethods.getConnectedEmulators = async function () {
+  try {
+    log.debug("Getting connected emulators");
+    let devices = await this.getConnectedDevices();
+    let emulators = [];
+    for (let device of devices) {
+      let port = this.getPortFromEmulatorString(device.udid);
+      if (port) {
+        device.port = port;
+        emulators.push(device);
+      }
+    }
+    log.debug(`${emulators.length} emulator(s) connected`);
+    return emulators;
+  } catch (e) {
+    log.errorAndThrow(`Error getting emulators. Original error: ${e.message}`);
```

Comment:
I'm starting to wonder whether this pattern (catching all errors and then wrapping them up in a catch block with a more descriptive error) is really what we want to do.

the value that it adds for the user is they get a message saying "error getting emulators" instead of something potentially harder to decipher.

is this worth it?